n3fit/src/evolven3fit/evolve.py

```diff
@@ -0,0 +1,204 @@
+    """
+    log_file = pathlib.Path(conf_folder) / "evolven3fit.log"
+    if log_file.exists():
+        raise SystemError("Log file already exists: evolven3fit has already been run?")
+    log_file = logging.FileHandler(log_file)
+    stdout_log = logging.StreamHandler(sys.stdout)
+    log_file.setLevel(logging.INFO)
+    stdout_log.setLevel(logging.INFO)
+    log_file.setFormatter(
+        logging.Formatter("%(asctime)s %(name)s/%(levelname)s: %(message)s")
+    )
+    stdout_log.setFormatter(
+        logging.Formatter("%(asctime)s %(name)s/%(levelname)s: %(message)s")
+    )
+    for logger_ in (log, *[logging.getLogger("eko")]):
+        logger_.handlers = []
+        logger_.setLevel(logging.INFO)
+        logger_.addHandler(log_file)
+        logger_.addHandler(stdout_log)
+    usr_path = pathlib.Path(conf_folder)
+    initial_PDFs_dict = load_fit(usr_path)
+    x_grid = np.array(
+        initial_PDFs_dict[list(initial_PDFs_dict.keys())[0]]["xgrid"]
+    ).astype(np.float)
+    theoryID = utils.get_theoryID_from_runcard(usr_path)
+    theory, op = eko_utils.construct_eko_cards(
+        theoryID, op_card_dict, t_card_dict, q_fin, q_points, x_grid
+    )
+    if eko_path is not None:
+        eko_path = pathlib.Path(eko_path)
+        log.info(f"Loading eko from : {eko_path}")
+        eko_op = output.Output.load_tar(eko_path)
+    else:
+        eko_op = eko_utils.construct_eko_for_fit(theory, op, log, dump_eko)
+    eko_op.xgrid_reshape(targetgrid=x_grid, inputgrid=x_grid)
+    info = gen_info.create_info_file(theory, op, 1, info_update={})
+    info["NumMembers"] = "REPLACE_NREP"
+    info["ErrorType"] = "replicas"
+    info["AlphaS_Qs"] = list(eko_op["Q2grid"].keys())
+    info["XMin"] = float(x_grid[0])
+    info["XMax"] = float(x_grid[-1])
+    dump_info_file(usr_path, info)
+    for replica in initial_PDFs_dict.keys():
+        evolved_block = evolve_exportgrid(initial_PDFs_dict[replica], eko_op, x_grid)
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+    # remove folder:
+    # The function dump_evolved_replica dumps the replica files in a temporary folder
+    # We need then to remove it after fixing the position of those replica files
+    (usr_path / "nnfit" / usr_path.stem).rmdir()
+
+
+def load_fit(usr_path):
+    """
+    Loads all the replica pdfs at fitting scale in usr_path and return the exportgrids
+
+    Parameters
+    ----------
+
+        usr_path: pathlib.Path
+            path to the folder containing the fit
+
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = usr_path / "nnfit"
+    replica_list = []
+    for yaml_file in nnfitpath.glob("replica_*/*.exportgrid"):
+        replica_list.append(yaml_file)
+    pdf_dict = {}
+    for replica in replica_list:
+        data = yaml.safe_load(replica.read_text())
+        pdf_dict[replica.parent.stem] = data
+    return pdf_dict
```

Comment:
I don't think you need the `replica_list`, you can directly loop over the `glob`.